cmake/projects/dlib/hunter.cmake

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2015, David Hirvonen
+# All rights reserved.
+
+if(DEFINED HUNTER_CMAKE_PROJECTS_DLIB_HUNTER_CMAKE_)
+  return()
+else()
+  set(HUNTER_CMAKE_PROJECTS_DLIB_HUNTER_CMAKE_ 1)
+endif()
+
+# Load used modules
+include(hunter_add_version)
+include(hunter_download)
+include(hunter_pick_scheme)
+include(hunter_add_package)
+
+# List of versions here...
+hunter_add_version(
+    PACKAGE_NAME
+    dlib
+    VERSION
+    "18.14"
+    URL
+    "https://github.com/davisking/dlib/archive/v18.14.tar.gz"
+    SHA1
+    85db9a8107cf5f65006921783164a0b2c0f3a5c0
+)
+
+# Pick a download scheme
+hunter_pick_scheme(DEFAULT url_sha1_no_install) # (for cmake modules and header-only libraries)
```

Comment:
Yes.  All the dlib files (mix of *.hpp and *.cpp) should be installed such
that files are accessible via the initial dlib path:

#include <dlib/image_processing.h>

#include <dlib/all/source.cpp>

There is a CMakeLists.txt and some bundled jpeg, png, and zlib code, but
that is only for building the sample applications and test code.

Some discussion here:

http://sourceforge.net/p/dclib/discussion/442517/thread/efc8d59d/

and here:

http://dlib.net/compile.html





On Wed, Apr 22, 2015 at 6:26 PM, Ruslan Baratov <notifications@github.com>
wrote:

> In cmake/projects/dlib/hunter.cmake
> <https://github.com/ruslo/hunter/pull/89#discussion_r28920961>:
>
> > +include(hunter_add_package)
> > +
> > +# List of versions here...
> > +hunter_add_version(
> > +    PACKAGE_NAME
> > +    dlib
> > +    VERSION
> > +    "18.14"
> > +    URL
> > +    "https://github.com/davisking/dlib/archive/v18.14.tar.gz"
> > +    SHA1
> > +    85db9a8107cf5f65006921783164a0b2c0f3a5c0
> > +)
> > +
> > +# Pick a download scheme
> > +hunter_pick_scheme(DEFAULT url_sha1_no_install) # (for cmake modules and header-only libraries)
>
> let's make config-mode from this package since it's header only library if
> must be a trivial task. What headers need to be installed? All '*.h' from
> dlib <https://github.com/davisking/dlib/tree/master/dlib> directory?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ruslo/hunter/pull/89/files#r28920961>.
>